### PR TITLE
VIMC-3862: Add endpoint for returning information about report parameters

### PR DESCRIPTION
**Pull request by r-ash**
Builds on #28, will be tidier looking diff after that one is merged.

### CODE REVIEW

File: R/api.R [hunk 1]
```diff
@@ -229,3 +230,34 @@ endpoint_available_reports <- function(runner) {
...
 }
+
+target_report_parameters <- function(runner, report_id, commit) {
+  tryCatch(
+    parameters <- runner$get_report_parameters(report_id, commit),
+    error = function(e) {
+      pkgapi::pkgapi_stop(e$message, "FAILED_RETRIEVE_PARAMS")
+    }
+  )
+  if (is.null(names(parameters)) && !is.null(parameters)) {
```
Comment (richfitz) [inline R/api.R:241]: Tiny thing but should the order here be reversed?

```
!is.null(parameters) && is.null(names(parameters))
```

as I started getting confused about why reports with no parameters were invalid. Or we could use an early exit:

```
if (is.null(parameters)) {
  return(list())
}
if (!is.null(names(parameters))) {
  # error
}
# rest of logic
```

File: inst/schema/ReportParameters.schema.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "id": "ReportParameters",
+    "type": "array",
+    "items": {
+        "name": "string",
+        "default": [ "string", "null" ]
```
Comment (richfitz) [inline inst/schema/ReportParameters.schema.json:7]: What does a default mean in this context?

File: R/api.R (hunk 1 above)
Comment (r-ash) [reply R/api.R]: Reversed the order of that, should hopefully be a bit clearer now.

File: inst/schema/ReportParameters.schema.json (hunk 2 above)
Comment (r-ash) [reply inst/schema/ReportParameters.schema.json:7]: The default value for the parameter, if one is set. I am taking this from
https://github.com/vimc/orderly/blob/0d4fd993e4ea03a0395edd07722662d8301c823f/tests/testthat/examples/parameters/src/example/orderly.yml

**Pull request merged**